sdk/src/fee.rs

```diff
@@ -92,10 +92,14 @@ impl FeeStructure {
     pub fn calculate_fee(
         &self,
         message: &SanitizedMessage,
-        _unused: u64,
+        for_test_only: u64,
         budget_limits: &FeeBudgetLimits,
         include_loaded_account_data_size_in_fee: bool,
     ) -> u64 {
+        if for_test_only == 0 {
+            return 0;
+        }
+
```

Comment:
This was the parameter's original function - to set entire fee to zero if it is zero

Reply:
this looks extremely hacky. can we just revert and try again, only correctly?

Reply:
This was exactly what it did originally - read `lamports_per_signature` from blockhash_queue or nonce, if it is zero, the multiple `0` to calculated fee (for test); otherwise multiple `1` to calculated fee. 

I was trying to get ride of this "for test only" parameter by initializing fee_structure from fee_rate_governor, which seems to not as reliable approach.

I'll revert the chain of PRs first. Then look for a better way to fix that weird parameter.

Reply:
> I'll revert the chain of PRs first. Then look for a better way to fix that weird parameter.

Are you planning to revert #34970 and #34865 or stick with this PR? I'm fine with either approach. If possible I'd like to release v1.18.1 today.

Reply:
If revert, also need to revert #34757, but I really want/need to keep its function. I am thinking either keep this PR (cause it's essentially what the reverted code does), or I can combine reverting #34865, #34757 and #34970, then an additional PR to do what #34757 on reverted codebase. @t-nelson are you have strong opinions on not to use this PR (even changing the var name to `test_fee_rate`, which is really what it is in current code base)?

Reply:
https://github.com/solana-labs/solana/pull/35016 reverts 3 PRs. imo, less better than this PR 😜

Reply:
I prefer this PR too... the new arg name and early return improve readability.